src/passes/FloatTrap.h

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
having a pass with a `.h` file feels wrong to me. a pass should be self-contained. how about a header file with the just the constants etc., probably under `ast/`?

for adding the pass to be run, can use the string name, instead of the class identifier.

Reply:
The pass needs an argument though, which means we need to have the Mode enum and constructors visible to callers.

It's weird and I'm not sure where to put it. It isn't registered by name with the pass system, so it doesn't really belong in `passes/`, but it is a pass over the module. Originally I had it all in a header file in `src/`, but that seemed wrong.
`ast/` seems fine to me, thanks.

Reply:
How about making separate passes for the modes: one pass would do the clamp mode, and one would do the JS mode? We can define them all in a single pass cpp file (see e.g. simplify-locals pass which defines 4).

Then the users would add the right pass of the 2 (or not add any pass if we allow traps). Might add a little utility function for that in the header.

Reply:
Yeah that's way cleaner, and has the benefit of letting us run this in `wasm-opt` as well, which means I can write some finer-grained tests for it.